packages/react/modules/components/configure-adapter/configure-adapter.tsx

```diff
@@ -42,91 +42,102 @@ type Props = {
   render?: () => React.ReactNode;
   children?: TConfigureAdapterChildren;
 };
-type State = {
-  appliedAdapterArgs: TAdapterArgs;
-};
-type AdapterState = valueof<AdapterStates>;
+type TAdapterState = valueOf<TAdapterStates>;
 
-// eslint-disable-next-line react/no-unsafe
-export default class ConfigureAdapter extends React.PureComponent<
-  Props,
-  State
-> {
-  static defaultProps = {
-    shouldDeferAdapterConfiguration: false,
-    defaultFlags: {},
-    children: null,
-    render: null,
-  };
+const ConfigureAdapter = (props: TProps) => {
+  const [appliedAdapterArgs, setAppliedAdapterArgs] = React.useState<
+    TAdapterArgs
+  >(props.adapterArgs);
+  const pendingAdapterArgs = React.useRef<TAdapterArgs | null>(null);
+  const adapterState = React.useRef<TAdapterState>(AdapterStates.UNCONFIGURED);
 
-  adapterState: AdapterState = AdapterStates.UNCONFIGURED;
-  pendingAdapterArgs?: TAdapterArgs | null = null;
+  const setAdapterState = React.useCallback(
+    (nextAdapterState: TAdapterState) => {
+      adapterState.current = nextAdapterState;
+    },
+    [adapterState]
+  );
 
-  state: { appliedAdapterArgs: TAdapterArgs } = {
-    appliedAdapterArgs: this.props.adapterArgs,
-  };
+  const applyAdapterArgs = React.useCallback(
+    (nextAdapterArgs: TAdapterArgs) => {
+      /**
+       * NOTE:
+       *   We can only unset `pendingAdapterArgs` after be actually perform
+       *   a batched `setState` otherwise outdated `adapterArgs` as we loose
+       *   the `pendingAdapterArgs` as we unset them too early.
+       */
+      setAppliedAdapterArgs(nextAdapterArgs);
+    },
+    []
+  );
 
-  setAdapterState = (nextAdapterState: AdapterState) => {
-    this.adapterState = nextAdapterState;
-  };
+  /**
+   * NOTE:
+   *   Clears the pending adapter args when applied adapter
+   *   args were set. Previously achived via `setState` callback.
+   */
+  React.useEffect(() => {
+    pendingAdapterArgs.current = null;
+  }, [appliedAdapterArgs]);
 
-  applyAdapterArgs = (nextAdapterArgs: TAdapterArgs) =>
-    /**
-     * NOTE:
-     *   We can only unset `pendingAdapterArgs` after be actually perform
-     *   a batched `setState` otherwise outdated `adapterArgs` as we loose
-     *   the `pendingAdapterArgs` as we unset them too early.
-     */
-    this.setState(
-      {
-        appliedAdapterArgs: nextAdapterArgs,
-      },
-      () => {
-        this.pendingAdapterArgs = null;
-      }
-    );
+  const getIsAdapterConfigured = React.useCallback(
```

Comment:
I am actually trying to move away from booleans in the code base. They seemsed apt but easily get hairy too. Cause you all of a sudden end up at 2-3 of them and then have to do the `if isA && isB || isC` stuff instead of enum style things which are a bit more like state machine states.

Reply:
Ah ok, so this is covered in the other PR?